src/main/java/util/TaskList.java

```diff
@@ -0,0 +1,147 @@
+package util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+import task.Task;
+
+/**
+ * Data structure to store tasks.
+ */
+public class TaskList {
+
+    private List<Task> tasks;
+
+    /**
+     * Creates new task list object with empty list of tasks.
+     */
+    public TaskList() {
+        tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Creates new task list object with the given list of tasks.
+     *
+     * @param tasks List of tasks to be stored.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns all tasks in a list format.
+     *
+     * @return List of all tasks.
+     */
+    public List<Task> getList() {
+        return tasks;
+    }
+
+    /**
+     * Prints all tasks sequentially.
+     */
+    public String printList() {
+
+        String output = "";
+
+        for (int i = 0; i < tasks.size(); i++) {
+            output += (i + 1) + ". " + tasks.get(i) + "\n";
+        }
+        if (output.length() == 0) {
+            output += "\n";
+        }
+
+        return output;
+    }
+
+    /**
+     * Prints all tasks sequentially that satisfied filter.
+     *
+     * @param filter Filters which tasks will be printed.
+     */
+    public String printList(Predicate<Task> filter) {
+
+        String output = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            if (filter.test(tasks.get(i))) {
+                output += (i + 1) + ". " + tasks.get(i) + "\n";
+            }
+        }
+
+        if (output.length() == 0) {
+            output += "\n";
+        }
+
+        return output;
+    }
```

Comment:
This is very similar to `printList()` and might be a violation of the DRY principle. Could the common logic be extracted to a helper function instead?

![image](https://user-images.githubusercontent.com/53135010/92477165-b1877880-f212-11ea-8700-ee0013115f9e.png)

Reply:
Or perhaps `printList()` could just point to `printList(x -> true)`.